docs/DeveloperGuide.md

```diff
@@ -1,38 +1,152 @@
+Concurrently, "add" command is implemented with behaviour to update CSV file to persist changes made to inventory.
+This command supports the following options:
+* '-h': Add custom fields to the inventory.
+* '-d': Add records to the inventory corresponding to defined fields.
+* '-hu': Updates fields in the inventory.
+
+#### Architecture Overview
+The 'AddCommand' is part of the `command` package, and interacts with the following key components:
+* **Inventory**: Contains the current state of the inventory, including fields and records.
+* **Csv**: Handles reading from and writing to the CSV file for data storage persistence.
+* **Ui**: Handles user interactions component such as message to user during command execution.
+
+Command flow of 'AddCommand' can be visualized with the following architecture diagram:
+![Architecture Diagram](link-to-architecture-diagram)
+
+
+#### Component-Level Design
+"AddCommand" Class:
+* processes the input arguments and executes the logic based on flags (-h, -d, -hu) respectively.
+* fields and records are validated before updating inventory data into the CSV file.
+
+"AddCommand" Methods:
+* handleAddMultipleFields(): processes input fields, validate fields, and update the inventory accordingly.
+* handleAddRecord(): handles logic for adding records into the inventory. If successfully add records, data will be appended into the CSV file.
+* handleUpdateFields(): allow updating of existing fields and types in the inventory.
+  ![Class Diagram](link-to-class-diagram)
+
+#### Sequence Diagram
+*Illustrates how "AddCommand" interacts with "Inventory" and "Csv" classes when adding a record:
+
+![Sequence Diagram](link-to-sequence-diagram)
+
+
+
+#### Why Implement "AddCommand" in this way?
+Separating implementation of "AddCommand", "Inventory", and "Csv" classes ensures:
+* "AddCommand" focus only on processing input and delegating task to other specific components.
+* "Csv" class is responsible for updating data into CSV file for storage; ensuring persistence in data control.
+* With this breakdown of implementation, create room for future scalability (e.g. adding more fields types).
+
+#### Alternative Considered
+Only when requirements permits, database implementation has been considered for better handling of fields types for inventory storage.
+
+### "Delete" Command Feature
+The "DeleteCommand" is responsible for adding fields and records to the inventory. 
+Concurrently, "delete" command is implemented with behaviour to update CSV file to persist changes made to inventory.
+This command supports the following formats:
+* '\<index>': Delete the record at the specified index (1-based indexing).
+* '-a': Delete all records in the inventory. 
+* '-e': Delete all records and headers in the inventory.
+* '-h \<field_name>': Delete the mentioned field and its column from the inventory.
+* '-r \<start>-\<end>': Delete records from the start index to the end index (both inclusive and 1-based indexing).
```

Comment:
Should have example below every command for clarity instead of in the manual testing part at the bottom of the document